File: backend/src/main/java/com/corporate/analytics/service/DepartmentService.java
package com.corporate.analytics.service;

import com.corporate.analytics.entity.Department;
import com.corporate.analytics.dto.request.CreateDepartmentRequest;
import com.corporate.analytics.dto.response.DepartmentResponseDTO;
import com.corporate.analytics.dto.request.UpdateDepartmentRequest;
import java.util.List;

public interface DepartmentService {

    List<Department> getAllDepartments();
    DepartmentResponseDTO createDepartment(CreateDepartmentRequest request);
    DepartmentResponseDTO getDepartmentById(Integer id);
    DepartmentResponseDTO updateDepartment(Integer id, UpdateDepartmentRequest request);
    void deleteDepartment(Integer id);
}

File: frontend/src/services/departmentService.js
import api from "./api";

export const getAllDepartments = async () => {

    const response = await api.get("/departments");

    return response.data;
};

export const createDepartment = async (departmentData) => {

    const response = await api.post(
        "/departments",
        departmentData
    );

    return response.data;

};

export const updateDepartment = async (departmentId, departmentData) => {

    const response = await api.put(
        `/departments/${departmentId}`,
        departmentData
    );

    return response.data;

};

export const deleteDepartment = async (departmentId) => {

    const response = await api.delete(`/departments/${departmentId}`);

    return response.data;

};
